src/main/java/teammates/common/datatransfer/attributes/AccountAttributes.java

```diff
@@ -26,49 +27,95 @@
     public Date createdAt;
     public StudentProfileAttributes studentProfile;
 
-    public AccountAttributes(Account a) {
-        googleId = a.getGoogleId();
-        name = a.getName();
-        isInstructor = a.isInstructor();
-        email = a.getEmail();
-        institute = a.getInstitute();
-        createdAt = a.getCreatedAt();
-        studentProfile =
-                a.getStudentProfile() == null ? null : StudentProfileAttributes.valueOf(a.getStudentProfile());
+    AccountAttributes() {
+        // attributes to be set after construction
     }
 
-    public AccountAttributes() {
-        // attributes to be set after construction
+    public static Builder builder() {
+        return new Builder();
     }
 
-    public AccountAttributes(String googleId, String name, boolean isInstructor,
-                String email, String institute, StudentProfileAttributes studentProfileAttributes) {
-        this.googleId = SanitizationHelper.sanitizeGoogleId(googleId);
-        this.name = SanitizationHelper.sanitizeName(name);
-        this.isInstructor = isInstructor;
-        this.email = SanitizationHelper.sanitizeEmail(email);
-        this.institute = SanitizationHelper.sanitizeTitle(institute);
-        this.studentProfile = studentProfileAttributes;
-        this.studentProfile.sanitizeForSaving();
+    public static class Builder {
+        private AccountAttributes accountAttributes;
 
-    }
+        public Builder() {
+            accountAttributes = new AccountAttributes();
+        }
+
+        public Builder withCreatedAt(Date createdAt) {
+            accountAttributes.createdAt = createdAt;
+            return this;
+        }
+
+        public Builder withStudentProfile(StudentProfile studentProfile) {
+            accountAttributes.studentProfile =
+                    studentProfile == null ? null : StudentProfileAttributes.valueOf(studentProfile);
+            return this;
+        }
+
+        public Builder withStudentProfileAttributes(StudentProfileAttributes studentProfileAttributes) {
+            accountAttributes.studentProfile = studentProfileAttributes;
+            accountAttributes.studentProfile.sanitizeForSaving();
+
+            return this;
+        }
+
+        public Builder withStudentProfileAttributes(String googleId) {
```

Comment:
Can we call it as `withDefaultStudentProfileAttributes` ? (As it actually uses the default (empty) profile with the provided  `googleId` right?)